core/src/main/java/org/apache/lucene/queryparser/classic/MapperQueryParser.java

```diff
@@ -779,7 +757,10 @@ private Query applySlop(Query q, int slop) {
                 builder.add(terms[i], positions[i]);
             }
             pq = builder.build();
-            pq.setBoost(q.getBoost());
+            //nocommit why do we deal with boost here? isn't that what applyBoost does?
```

Comment:
This method takes a phrase query and is supposed to create an equivalent phrase query that just has a different value of the slop, so we need to transfer the boost?

Maybe the method should look like this now:

```java
    private Query applySlop(Query q, int slop) {
        float boost = 1f;
        Query underlyingQuery = q;
        while (underlyingQuery instanceof BoostQuery) {
            BoostQuery bq = (BoostQuery) underlyingQuery;
            boost *= bq.getBoost();
            underlyingQuery = bq.getQuery();
        }
        if (underlyingQuery instanceof PhraseQuery) {
            PhraseQuery pq = (PhraseQuery) underlyingQuery;
            PhraseQuery.Builder builder = new PhraseQuery.Builder();
            builder.setSlop(slop);
            final Term[] terms = pq.getTerms();
            final int[] positions = pq.getPositions();
            for (int i = 0; i < terms.length; ++i) {
                builder.add(terms[i], positions[i]);
            }
            pq = builder.build();
            pq.setBoost(boost);
            return pq;
        } else if (underlyingQuery instanceof MultiPhraseQuery) {
            ((MultiPhraseQuery) underlyingQuery).setSlop(slop);
            return q;
        } else {
            return q;
        }
    }
```